ci/tests/smash_test.c

```diff
@@ -0,0 +1,41 @@
+/* Test that overwrite error detection works reasonably.        */
+
+#ifndef GC_DEBUG
+# define GC_DEBUG
+#endif
+
+#include "gc.h"
+
+#include <stdio.h>
+
+#define COUNT 7000
+#define SIZE  40
+
+char * A[COUNT];
+
+char * volatile q;
+
+int main(void)
+{
+  int i;
+  char *p;
+
+  GC_INIT();
+
+  for (i = 0; i < COUNT; ++i) {
+     A[i] = p = (char*)GC_MALLOC(SIZE);
+
+     if (i%3000 == 0) {
+        q = NULL;
+        GC_gcollect();
+     } else if (i%5678 == 0 && p != 0) {
```

Comment:
I'm not sure I understand the 5678 or 3000 constants. Can they be defined in terms of (an appropriate) word size?

Reply:
Aren't they just (arbitrary) frequency settings for the test? I notice that `i%5678` only fires once, since `COUNT` is `7000` (and the zero case is caught in the `if`).

Reply:
Are we copying it only because we can't currently run the *whole* test suite?

Reply:
> Are we copying it only because we can't currently run the _whole_ test suite?

Partially yes. But also because we did not want to add our CI scripts to the original bdwgc test directory.

Reply:
> Aren't they just (arbitrary) frequency settings for the test? I notice that `i%5678` only fires once, since `COUNT` is `7000` (and the zero case is caught in the `if`).

I thought so too - but not sure.

Reply:
> But also because we did not want to add our CI scripts to the original bdwgc test directory.

I don't think that should be true. We can execute whatever we want from `.buildbot.sh` so if we can execute the original test(s) without copying the source, so much the better!

Reply:
I agree. It was just easier to put all our stuff in a separate directory until we get the whole set of tests working. In the fullness of time, hopefully this directory will be redundant.

Reply:
Instead of copying tests, can we either a) symlink them in `.buildbot.sh` or b) just compile them individually? That'll be much better than having some tests that might have diverged from the originals!

Reply:
Done option **(a)** in  [db4a02b](https://github.com/capablevms/bdwgc/pull/15/commits/db4a02be281005c9f9cb8b83b287674ed22a3817).

Reply:
I like it!